package/MDAnalysis/core/selection.py

```diff
@@ -979,6 +1012,7 @@ def __init__(self, parser, tokens):
             raise ValueError(
                 "Invalid operator : '{0}' Use one of : '{1}'"
                 "".format(oper, self.ops.keys()))
+        self.value = float(value)
```

Comment:
But yeah, we'll do it in another PR, I think getting the AndOperation working right will be a small headache